### PR TITLE
Add custom option that allows to sort branches

### PR DESCRIPTION
**Pull request by ivaaaan**
When I checkout, I want to see branches sorted by commit date. 
This change allows user to specify their own sort settings.

Flag description: https://git-scm.com/docs/git-branch#Documentation/git-branch.txt---sortltkeygt
Sorting keys: https://git-scm.com/docs/git-for-each-ref#_options

#144 #392

### CODE REVIEW

File: lua/neogit/config.lua [hunk 1]
```diff
@@ -9,6 +9,7 @@ M.values = {
   disable_insert_on_commit = true,
   use_magit_keybindings = false,
   auto_refresh = true,
+  sort_branches = "-commiterdate",
```
Comment (ivaaaan) [inline lua/neogit/config.lua:12]: Maybe this one should be changed to `nil`/something else.

**Comment by ten3roberts** [comment]
This looks fantastic!

Is there anything more you'd like to add or update, since it has been some time you implemented the feature and it may thus need some tweaking?

Otherwise, I am more than happy to merge

**Review by ten3roberts — APPROVED** [review]

**Comment by ivaaaan** [comment]
Thanks for the review :) Not really, the only thing I don't like is that it shows remote branches next to the local branches. I would hide remote branches, but not sure how it will affect others. 
image

**Comment by asymmetric** [comment]
Maybe documenting this could be a good idea? Or is this self-documenting for end-users through the UI somehow?

**Comment by ten3roberts** [comment]
> Maybe documenting this could be a good idea? Or is this self-documenting for end-users through the UI somehow?

Agreed, perhaps a mention in the readme would suffice, otherwise it is quite self documenting.

**Comment by ten3roberts** [comment]
Closes: #392

**Comment by ivaaaan** [comment]
I have updated  README

**Review by ten3roberts — APPROVED** [review]

**Comment by ten3roberts** [comment]
I can merge this after we fix the stylua linting.

Thanks for the PR, as this is something I've been needing a long time

**Comment by ivaaaan** [comment]
@ten3roberts linting seems to be fine now, at least locally :) 
Thanks for your work.

**Comment by timsofteng** [comment]
@ivaaaan may you descibe please how to sort branches by last checkout as like lazygit do?

**Pull request merged**